prover_rust/print_halo2gpu_version.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+config_file="$HOME/.cargo/config"
+
+if [ ! -e "$config_file" ]; then
+    exit 0
```

Comment:
print error before exit

Reply:
you can also borrow some codes from https://github.com/scroll-tech/testnet/blob/mainnet/core/contracts/scripts/common.sh

Reply:
This script is copied from go version's prover, and it's used in Makefile to bundle the halo2_gpu commit info into the prover_rust binary. If there is some error in the script, we can detect it by seeing ./prover_rust --version.
I suggest not changing this now..